include/forcing/Forcing.h

```diff
@@ -102,23 +139,30 @@ class Forcing
      */
     double get_current_hourly_precipitation_meters_per_second()
     {
-        return precipitation_rate_meters_per_second_vector[forcing_vector_index_ptr];
+        check_forcing_vector_index_bounds();
+
+        return precipitation_rate_meters_per_second_vector.at(forcing_vector_index);
     }
 
     /**
      * @brief Gets next hourly precipitation in meters per second
      * Increments pointer in forcing vector by one timestep
      * Precipitation frequency is assumed to be hourly for now.
-     * TODO: Add input for dt (delta time) for different frequencies in the data than the model frequency.
+     * /// \todo: Add input for dt (delta time) for different frequencies in the data than the model frequency.
+     * /// \todo: Reconsider incrementing the forcing_vector_index because other functions rely on this, and it
+     *            could have side effects
      * @return the current hourly precipitation in meters per second after pointer is incremented by one timestep
      */
     double get_next_hourly_precipitation_meters_per_second()
     {
-        //Increment forcing index
-        forcing_vector_index_ptr = forcing_vector_index_ptr + 1;
+        //Check forcing vector bounds before incrementing forcing index
+        if (forcing_vector_index < precipitation_rate_meters_per_second_vector.size() - 1)
```

Comment:
This is off by 1. Should be `forcing_vector_index < precipitation_rate_meters_per_second_vector.size()` if using strict inequality.

Reply:
This is actually a check before incrementing the index as a bounds check to what would be after incrementing. Do you think this is correct given that case then?

Reply:
Yup, you are correct.  Disregard.